### ISSUE TITLE
Unable to import legacy suite

### ISSUE THREAD

**Issue by Elbap** [SeleniumHQ/selenium-ide#224]
I have been trying to import a legacy test suite, I have tried through FF and Chrome but Selenium IDE just hangs, it doesn't do anything. Please advise.

**Comment by corevo** [comment]
Can you tell me if theres an error in the console?  
Maybe if possible upload your test cases

**Comment by mnbeer** [comment]
Not sure if this is same issue but dragging a file in generates a check mark and a class named "accepted" is associated with the appropriate li tag. However, it is not clear where to go from there. Only visual options are "Cancel" and close dialog.

![image](https://user-images.githubusercontent.com/7638116/41123729-7c5a4d24-6a54-11e8-89ae-3934e522c9ce.png)

**Comment by 98gmarquee** [comment]
Wondering if you had the same issue I did. The new IDE requires that the name inside the title field in the html is unique. I have a nasty habit of creating one script,then copying it. The old ide would just show the file name in the suite section and never really cared about the actual <title> field in the html. The new one does care. 

I just ran into an issue not being able to import my suite. I opened the suite, it gave me the dialog, then I dragged my html's over to it and it just sat there and blinked at me. Took me a few minutes to remember having this issue a while back.

I then changed all my test files to have the file prefix match the title tag in the file, then it imported fine doing the above steps.

**Comment by Dukuu-Wild** [comment]
While importing a testsuite i also get a blank SeIDE window, as it hangs.
But than i tried to import one-by-one, and appeared there are faulty testcases there.
Most of them have comments with  problem.
But one cannot know what is the problem: when one gives all the files wanted for testsuite it just hangs. @corevo does it helps?

**Comment by corevo** [comment]
@Dukuu-Wild that problem is tracked at #237

**Comment by Dukuu-Wild** [comment]
Please check also the following cases:
1. While importing the testsuite with multiple same testcases used, it assumes that same-named testcases should be different files. See the screenshot: http://i.prntscr.com/cE6dEwTsROCF3VJA1APtMQ.png
Red lines should link to the same file that is already linked.
Somewhy this multi-linkage is not respected.
Here are sample files:  
[Suite login_multiple cases.zip](https://github.com/SeleniumHQ/selenium-ide/files/2111289/Suite.login_multiple.cases.zip)

2. When there is a faulty case there in testsuite opened, the SeIDE hangs blank:
(case with "!" in name has comment problem from #237 )
http://i.prntscr.com/eCpYIzlyRV_32g_av2TFJw.png
After accepting this dialog it hangs blank.
Sample files: [Suite login_faulty case in suite.zip](https://github.com/SeleniumHQ/selenium-ide/files/2111318/Suite.login_faulty.case.in.suite.zip)

3. The sequence of cases from testsuite is not respected:
Screenshot http://i.prntscr.com/gESbJ0tqS6CoLRtv3dEpFg.png
Sample files: 
[Suite login_sequence of cases in suite.zip](https://github.com/SeleniumHQ/selenium-ide/files/2111314/Suite.login_sequence.of.cases.in.suite.zip)

**Comment by corevo** [comment]
I'll take a look at point 1 & 2.  
About point 3, tests in the general test view are sorted alphabetically, #128 will let you sort them inside a test suite.

**Comment by Dukuu-Wild** [comment]
Alphabetical order in suite is an obvious pain and cause unwanted duplicating of testcases in testsuites.
I've read that branch #128. Great thing that it is going to be implemented. And yes, pt.3 is blocked by that. But this is still an issue to check after #128 is done, as we talk about importing procedure here.
Import should be done not alphabetically but in the same way it is listed in a testsuite-file.
Thanks in forward.
Great thanks for your work.

**Comment by corevo** [comment]
@Dukuu-Wild fixed as 58c6e42e.  
Please note when importing that only one copy of the test suite will be imported, this is the intended behavior, instead of having users have multiple copies of the same test case, we are working on a way to invoke the same test case multiple times.  
I know that this isn't solving the issue completely, as the test case would only run once, the issue of running the test case multiple times will be tracked at #244 as it is a more focused issue.

**Comment by corevo** [comment]
Closed since there is no response from OP.  
@Elbap please answer my first comment to reopen this issue.

**Comment by Dukuu-Wild** [comment]
@corevo actually case#2 from my comment above makes the same behavior as described in the start of this thread. Did you try it?

**Comment by corevo** [comment]
Yes, and I'm working on it.
But I'm not sure if that's the original cause of the issue.

Cheers,
Tom

**Comment by yawningemu** [comment]
Any updates on this issue?, console does not show any errors for me

**Comment by corevo** [comment]
It is resolved.

**Comment by yawningemu** [comment]
I still face the same issue

**Comment by corevo** [comment]
I just tried the suite you attached as 2. and it worked perfectly for me.

**Comment by yawningemu** [comment]
Suites are working fine on my end too, its the individual test cases that I am not able to import

**Comment by corevo** [comment]
I was able to import them manually one-by-one.  
Add steps to reproduce this with the faulty test case please.

**Comment by yawningemu** [comment]
**I am using another tool to create test cases and then exporting them as suite**
[faulty.zip](https://github.com/SeleniumHQ/selenium-ide/files/2179805/faulty.zip)
I cannot import the items as either a suite or individual test cases
Though I was able to import @Dukuu-Wild suite, maybe because his suite contains an additional file

**Comment by corevo** [comment]
I can import both cases you attached

**Comment by Dukuu-Wild** [comment]
Ahhhaha, i found why @yawningemu can't import cases =)))
1. It's not possible to open testcases themselves - only testsuites.
2. htm**l** are not accepted, only htm 
3. So i concatenated my suite with yawningemu's "theStore" case, that i renamed to htm extension and successfully imported the suite with cases.
[Suite login_faulty case in suite+store.zip](https://github.com/SeleniumHQ/selenium-ide/files/2180617/Suite.login_faulty.case.in.suite.store.zip)

As for me what we have to consider as ToDo:
1. Make the imported files not critical to extension: let it be htm, html or no extension at all 
2. Implement the way to import testcases without a suite, so one can open them without testsuites. As a separate button or in same button.
@corevo what do you think about it? Is it possible? Shall we split this task to other ticket?

**Comment by corevo** [comment]
You can open them without the suite, they will be added as another test case.  
I'll make sure to include both `htm` and `html` as sources, if the problem persists open a new ticket, we are diverging from the issue the op has reported.

**Comment by corevo** [comment]
Fixed in 3ff7c793
